### ISSUE TITLE
Ubuntu install libyaml-dev

### ISSUE THREAD

**Issue by MSP-Greg** [actions/runner-images#6725]
### Tool name

libyaml-dev

### Tool license

apt package

### Add or update?

- [X] Add
- [ ] Update

### Desired version

na - use current Ubuntu pacakge

### Approximate size

small

### Brief description of tool

Ubuntu has libyaml installed (automatic), but not libyaml-dev.  I believe macOS has it installed, as it does have libyaml header files, etc.

Ignoring Windows, as it depends on the build system in use.

Reason for the request is Ruby's yaml package (`psych`) previously vendored libyaml code.  With the new release that was removed.  It's a package that is a dependency for many popular Ruby gems, and now CI is failing on Ubuntu, but fine on macOS.  It seems reasonable to ask for parity between the two...

### Provide a basic test case to validate the tool's functionality.

```bash
apt install libyaml-dev
```

### Platforms where you need the tool

- [ ] Azure DevOps
- [ ] GitHub Actions

### Runner images where you need the tool

- [ ] Ubuntu 18.04
- [X] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Can this tool be installed during the build?

```bash
Yes.
```

### Tool installation time in runtime

quick

### Are you willing to submit a PR?

not sure if there is a generic 'install these apt packages' list

**Comment by erik-bershel** [comment]
Hi @MSP-Greg!
We'll take a closer look and return with decision a bit later.

**Comment by MSP-Greg** [comment]
@erik-bershel

I've been involved in with setup-ruby since its inception.  I've also been involved in the Windows code it uses, which does install a full set of build tools and/or packages (MSYS2 mingw, ucrt, or msft/vcpkg), so we do install/update build systems in its code.  The code also caches all the packages in GitHub releases, so it's not hammering external package servers.

We've never needed to do so with Ubuntu or macOS, and most Ruby CI probably has more Ubuntu & macOS jobs than Windows.

Summing up, after looking at all the practical aspects of this issue, adding it to the images seems like the best solution.

**Comment by eregon** [comment]
To give a bit more context, this is coming from https://github.com/ruby/setup-ruby/issues/409
Here is a summary:
Ruby added a dependency on libyaml and the `psych` gem (part of Ruby) added a dependency libyaml-dev (the headers), due to no longer vendoring libyaml in `psych`.
This means a large number of Ruby-related workflow started to fail when installing the `psych` gem (which is a frequent transitive dependency) since psych 5 was released with that change.

So it would be nice and useful if runner images include `libyaml-dev` to avoid many many workflows to have to be updated to `sudo apt install libyaml-dev` on Ubuntu.
It's just a few headers, about ~250 KB installed (from https://packages.debian.org/stretch/libyaml-dev).

> Tool installation time in runtime

Actually it's something like 15s so it's non-trivial overhead (it's ~3x slower than downloading and setting up Ruby to give an idea).

**Comment by mikhailkoliada** [comment]
will be deployed next week

**Comment by mikhailkoliada** [comment]
Deployed
